pom.xml

```diff
@@ -76,5 +76,10 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
```

Comment:
Yes, by contract the maps should be immutable. Even if this weren't the case, I like these builders because they are much more expressive than the built-in HashMap and HashSet. Compare:
```java
return ImmutableMap.<String, Metric>builder()
        .put("activeThreadCount", (Gauge<Integer>) pool::getActiveThreadCount)
        .put("poolSize", (Gauge<Integer>) pool::getPoolSize)
        .put("queuedSubmissionCount", (Gauge<Integer>) pool::getQueuedSubmissionCount)
        .put("runningThreadCount", (Gauge<Integer>) pool::getRunningThreadCount)
        .put("queuedTaskCount", (Gauge<Long>) pool::getQueuedTaskCount)
        .put("stealCount", (Gauge<Long>) pool::getStealCount)
        .build();
```
with
```java
Map<String, Metric> metrics = new HashMap<>();
metrics.put("activeThreadCount", (Gauge<Integer>) pool::getActiveThreadCount);
metrics.put("poolSize", (Gauge<Integer>) pool::getPoolSize);
metrics.put("queuedSubmissionCount", (Gauge<Integer>) pool::getQueuedSubmissionCount);
metrics.put("runningThreadCount", (Gauge<Integer>) pool::getRunningThreadCount);
metrics.put("queuedTaskCount", (Gauge<Long>) pool::getQueuedTaskCount);
metrics.put("stealCount", (Gauge<Long>) pool::getStealCount);
return Collections.unmodifiableMap(metrics);
```
Maybe you could elaborate on your concern?